src/bias/MetaD.cpp

```diff
@@ -1592,73 +1701,131 @@ double MetaD::evaluateGaussian(const vector<double>& cv, const Gaussian& hill, d
         }
       }
     }
+  } else {
+    for(unsigned i=0; i<ncv; i++) {
+      double dp=difference(i,hill.center[i],pcv[i])*hill.invsigma[i];
+      dp2+=dp*dp;
+    }
+    dp2*=0.5;
+  }
+
+  double bias=0.0;
+  if(dp2<DP2CUTOFF) bias=hill.height*std::exp(-dp2);
+
+  return bias;
+}
+
+double MetaD::evaluateGaussianAndDerivatives(const std::vector<double>& cv, const Gaussian& hill, std::vector<double>& der, std::vector<double>& dp_)
```

Comment:
@carlocamilloni is `dp_` here a local variable? I ask because it looks to me that when you call this function you do not use the modified `dp_`. If so it would suggest to:
- declare it as local for clarity
- for optimization, you could declare it as `std::array<double> dp[maxdim]`

In `tools/Grid.h`  we already set maxdim to some very high value (I think 64). This allows to avoid the dynamic allocation, should be faster

Reply:
@GiovanniBussi sorry I am not sure If I  understand your suggestion. dp_ is local in the sense that is only used inside evaluateGaussianAndDerivatives, but it needs to be defined outside the function because doing so allow to reduce by 50% the calculation time spent for that, indeed before it was defined within the class, but that was not thread-safe. I played a bit with different implementation, the current one is not elegant but is the fastest. I tried now using array inside evaluate and it is faster than vector but still slower than the current implementation

Reply:
Yes indeed I was suggesting something like:
```
double MetaD::evaluateGaussianAndDerivatives(const std::vector<double>& cv, const Gaussian& hill, std::vector<double>& der) {
  std::array<double> dp_;
  ...
```

I am a bit surprised it is slower than having a `std::vector` declared outside, and passed as an argument, but it is possible that these small vectors are well optimized. So, if it's faster, just leave it as you did it :-)

Reply:
(sorry the syntax was incorrect , it should be something like `  std::array<double,maxsize> dp_;`)